src/lightning/fabric/plugins/precision/deepspeed.py

```diff
@@ -77,8 +77,10 @@ def init_context(self) -> Generator[None, None, None]:
 
         default_dtype = torch.get_default_dtype()
         torch.set_default_dtype(self._desired_dtype)
-        yield
-        torch.set_default_dtype(default_dtype)
+        try:
+            yield
+        finally:
+            torch.set_default_dtype(default_dtype)
```

Comment:
should we factor this out to a utility function so that we don't accidentally forget the finally part somewhere in the future?

Reply:
Working on it in https://github.com/Lightning-AI/lightning/pull/18557